### PR TITLE
replace AdvanceSubpass with Begin/EndRenderSubpass

### PR DESCRIPTION
**Pull request by kainino0x**
This change is motivated by usage transitions: if there are usage transitions in between subpasses, AdvanceSubpass makes it unnatural. Plus, if we go the route of all-explicit attachment transitions, they have to be done between subpasses.

TODO:

* [x] Should the first BeginRenderSubpass and the last EndRenderSubpass be implicit? A little asymmetrical, but it doesn't seem like a problem. - decided not to do this for now
* [x] Fix D3D12 code (still a no-op on D3D12 backend)

### CODE REVIEW

File: src/backend/common/CommandBufferStateTracker.cpp [hunk 1]
```diff
@@ -164,10 +164,6 @@ namespace backend {
     };
```
Comment (Kangz) [inline src/backend/common/CommandBufferStateTracker.cpp]: A new validation aspect should be added "SUBPASS" which should gate most operations that are currently gated on "RENDERPASS"

**Comment by Kangz** [comment]
Looks good with the validation changes.

File: src/backend/common/CommandBufferStateTracker.cpp (hunk 1 above)
Comment (kainino0x) [inline src/backend/common/CommandBufferStateTracker.cpp]: Actually currently most things are gated on RENDER_PIPELINE, we don't have a RENDERPASS validation aspect. I'm not sure we need a SUBPASS validation aspect (at least, not right at the moment) - I'm not spotting any area where the current validation is wrong.

Comment (kainino0x) [inline src/backend/common/CommandBufferStateTracker.cpp]: (subpassActive could probably be replaced with an aspect though)

Comment (Kangz) [inline src/backend/common/CommandBufferStateTracker.cpp]: Makes thanks, LGTM

**Comment by kainino0x** [comment]
Replaced subpassActive with a validation aspect (and rebased). PTAL and merge if ready.

**Pull request merged**